File: utilities/src/main/java/com/toocol/ssh/utilities/utils/Asable.java
package com.toocol.ssh.utilities.utils;

/**
 * @author ：JoeZane (joezane.cn@gmail.com)
 * @date: 2022/6/20 23:39
 * @version: 0.0.1
 */
public interface Asable {

    @SuppressWarnings("all")
    default <T> T as() {
        return (T)this;
    }

}

File: samples/coroutines/src/main/java/com/toocol/coroutines/core/Asable.kt
package com.toocol.coroutines.core

/**
 * @author ：JoeZane (joezane.cn@gmail.com)
 * @date: 2022/6/20 23:39
 * @version: 0.0.1
 */
interface Asable {
    fun <T> `as`(): T {
        return this as T
    }
}